constancecchen/kibana
# Implement remaining unit tests

## https://github.com/elastic/app-search-team/issues/766

- Writes all remaining unit tests for the Enterprise Search plugin
  - Adds a few extra tests to existing tests that were missing statement/branch coverage
- Major test refactors:
  - Adds a test helper to enable use of `useContext`/a mocked `KibanaContext` from Enzyme's `shallow()` wrapper (634ae14) (example usage: afa53e6 and 077b1e1)
  - Adds a test helper that mocks React Router's history/useHistory (b84ffb5)

There's a lot going on, so I definitely recommend [following along by commit](https://github.com/constancecchen/kibana/pull/7/commits) if possible 🙇‍♀️ 

## QA

<img width="1438" alt="" src="https://user-images.githubusercontent.com/549407/80750610-3af8a280-8add-11ea-9b49-d2d8e273c0ee.png">

- `cd` or ensure you're in the `kibana/x-pack` folder
- `yarn test:jest plugins/enterprise_search --coverage` from the x-pack folder
- To view the coverage report (see above screenshot), go to `kibana/target/kibana-coverage/jest/index.html` from any browser.
